cluster/query_executor_test.go

```diff
@@ -0,0 +1,300 @@
+			}}, nil
+		}
+		return &ic
+	}
+
+	// Verify all results from the query.
+	if a := ReadAllResults(e.ExecuteQuery(`SELECT count(value) FROM cpu`, "db0", 0)); !reflect.DeepEqual(a, []*influxql.Result{
+		{
+			StatementID: 0,
+			Series: []*models.Row{{
+				Name:    "cpu",
+				Columns: []string{"time", "count"},
+				Values: [][]interface{}{
+					{time.Unix(0, 0).UTC(), float64(30)},
+				},
+			}},
+		},
+	}) {
+		t.Fatalf("unexpected results: %s", spew.Sdump(a))
+	}
+}
+
+// QueryExecutor is a test wrapper for cluster.QueryExecutor.
+type QueryExecutor struct {
+	*cluster.QueryExecutor
+
+	MetaClient MetaClient
+	TSDBStore  TSDBStore
+	LogOutput  bytes.Buffer
+}
+
+// NewQueryExecutor returns a new instance of QueryExecutor.
+// This query executor always has a node id of 0.
+func NewQueryExecutor() *QueryExecutor {
+	e := &QueryExecutor{
+		QueryExecutor: cluster.NewQueryExecutor(),
+	}
+	e.Node = &influxdb.Node{ID: 0}
+	e.QueryExecutor.MetaClient = &e.MetaClient
+	e.QueryExecutor.TSDBStore = &e.TSDBStore
+
+	e.QueryExecutor.LogOutput = &e.LogOutput
+	if testing.Verbose() {
+		e.QueryExecutor.LogOutput = io.MultiWriter(e.QueryExecutor.LogOutput, os.Stderr)
+	}
+
+	return e
+}
+
+// DefaultQueryExecutor returns a QueryExecutor with a database (db0) and retention policy (rp0).
+func DefaultQueryExecutor() *QueryExecutor {
+	e := NewQueryExecutor()
+	e.MetaClient.DatabaseFn = DefaultMetaClientDatabaseFn
+	e.TSDBStore.ExpandSourcesFn = DefaultTSDBStoreExpandSourcesFn
+	return e
+}
+
+// ExecuteQuery parses query and executes against the database.
+func (e *QueryExecutor) ExecuteQuery(query, database string, chunkSize int) <-chan *influxql.Result {
+	return e.QueryExecutor.ExecuteQuery(MustParseQuery(query), database, chunkSize, make(chan struct{}))
+}
+
+// TSDBStore is a mockable implementation of cluster.TSDBStore.
+type TSDBStore struct {
+	CreateShardFn  func(database, policy string, shardID uint64) error
+	WriteToShardFn func(shardID uint64, points []models.Point) error
+
+	DeleteDatabaseFn                func(name string) error
+	DeleteMeasurementFn             func(database, name string) error
+	DeleteRetentionPolicyFn         func(database, name string) error
+	DeleteSeriesFn                  func(database string, sources []influxql.Source, condition influxql.Expr) error
+	ExecuteShowFieldKeysStatementFn func(stmt *influxql.ShowFieldKeysStatement, database string) (models.Rows, error)
+	ExecuteShowSeriesStatementFn    func(stmt *influxql.ShowSeriesStatement, database string) (models.Rows, error)
+	ExecuteShowTagValuesStatementFn func(stmt *influxql.ShowTagValuesStatement, database string) (models.Rows, error)
+	ExpandSourcesFn                 func(sources influxql.Sources) (influxql.Sources, error)
+	ShardIteratorCreatorFn          func(id uint64) influxql.IteratorCreator
+}
+
+func (s *TSDBStore) CreateShard(database, policy string, shardID uint64) error {
+	return s.CreateShardFn(database, policy, shardID)
+}
```

Comment:
The spacing between functions here is very weird. There should probably be a newline between each function just for style.